src/test/scala/org.openeo/HttpCache.scala

```diff
@@ -34,16 +38,30 @@ class HttpCache extends sun.net.www.protocol.https.Handler {
           val path = Paths.get(cachePath, filePath)
           if (!Files.exists(path)) {
             println("Caching request url: " + url)
-            Files.createDirectories(Paths.get(cachePath, basePath))
-            val stream = openConnectionSuper(url).getInputStream
-            val tmpBeforeAtomicMove = Paths.get(cachePath, java.util.UUID.randomUUID().toString)
-            Files.copy(stream, tmpBeforeAtomicMove)
-            Files.move(tmpBeforeAtomicMove, path)
+            try {
+              Files.createDirectories(Paths.get(cachePath, basePath))
+              val stream = openConnectionSuper(url).getInputStream
+              val tmpBeforeAtomicMove = Paths.get(cachePath, "unconfirmed_download_" + UUID.randomUUID())
+              Files.copy(stream, tmpBeforeAtomicMove)
+              Files.move(tmpBeforeAtomicMove, path)
```

Comment:
Since we take full control over the super's inputstream we should also close the original stream after writing its data to a file.

Reply:
true, wrapped in a try-finaly